### DISCUSSION TITLE
Celery worker deletes SQS message as soon as it's received, before processing

### DISCUSSION THREAD

**Discussion by Lucianod28** [celery/celery#9452] [Q&A] [answered]
Hi, I have a scheduled long-running celery task defined like so:
```python
"create-automated-position-report": {
    "task": "create_automated_position_report",
    "schedule": timezone.timedelta(minutes=30),
    "options": {"MessageGroupId": "create_automated_position_report"},
}
```
I ran into an issue where when this tasks takes more than 30 minutes, multiple instances of it run concurrently, despite using a FIFO SQS queue. After investigating, I found that the task is deleted by the worker as soon as it's received, before it's processed. Why is this happening?

These are the full worker logs from when the message was received to deleted: [extract-2024-12-05T18_42_38.448Z.csv](https://github.com/user-attachments/files/18028193/extract-2024-12-05T18_42_38.448Z.csv)

I'm using Celery 5.4.0 and Kombu 5.4.2.

**Comment by mike-optifino** [comment, marked as answer]
HI! Please check what value has [this parameter](https://docs.celeryq.dev/en/stable/userguide/tasks.html#Task.acks_late)

**Reply by Lucianod28** [reply]
Thanks, setting that fixed it. For some reason I thought celery deletes the message after completion by default.
